spec/insurance_spec.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe EasyPost::Insurance do
+  describe '.create' do
+    it 'creates an insurance object' do
+      shipment = EasyPost::Shipment.create(Fixture.one_call_buy_shipment)
+
+      insurance = described_class.create(
+        {
+          to_address: Fixture.basic_address,
+          from_address: Fixture.basic_address,
+          tracking_code: shipment.tracking_code,
+          carrier: Fixture.usps,
+          amount: '100',
+        },
+      )
+
+      expect(insurance).to be_an_instance_of(described_class)
+      expect(insurance.id).to match('ins_')
+    end
+  end
+
+  describe '.retrieve' do
+    it 'retrieves an insurance object' do
+      shipment = EasyPost::Shipment.create(Fixture.one_call_buy_shipment)
+
+      insurance = described_class.create(
+        {
+          to_address: Fixture.basic_address,
+          from_address: Fixture.basic_address,
+          tracking_code: shipment.tracking_code,
+          carrier: Fixture.usps,
+          amount: '100',
+        },
+      )
+
+      retrieved_insurance = described_class.retrieve(insurance.id)
+
+      expect(retrieved_insurance).to be_an_instance_of(described_class)
+      expect(insurance.id).to eq(retrieved_insurance.id)
```

Comment:
I think the order is wrong here, correct me if I am wrong.
```suggestion
      expect(retrieved_insurance.id).to eq(insurance.id)
```

Reply:
Nice catch